mlflow/gateway/utils.py

```diff
@@ -259,3 +261,45 @@ def is_valid_mosiacml_chat_model(model_name: str) -> bool:
 
 def is_valid_ai21labs_model(model_name: str) -> bool:
     return model_name in {"j2-ultra", "j2-mid", "j2-light"}
+
+
+def strip_sse_prefix(s: str) -> str:
+    # https://html.spec.whatwg.org/multipage/server-sent-events.html
+    return re.sub(r"^data:\s+", "", s)
+
+
+def to_sse_chunk(data: str) -> str:
+    # https://html.spec.whatwg.org/multipage/server-sent-events.html
+    return f"data: {data}\n\n"
+
+
+def _find_boundary(buffer: bytes) -> int:
+    try:
+        return buffer.index(b"\n")
+    except ValueError:
+        return -1
+
+
+async def handle_incomplete_chunks(
+    stream: AsyncGenerator[bytes, Any]
+) -> AsyncGenerator[bytes, Any]:
+    """Wraps a streaming response and handles incomplete chunks from the server."""
```

Comment:
can we include `https://community.openai.com/t/incomplete-stream-chunks-for-completions-api/383520` in the docstring?

Reply:
Sure